### PR TITLE
[folly] Add folly v2023.09.25.00

### PR DESCRIPTION
**Pull request by jjerphan**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: versions/f-/folly.json [hunk 1]
```diff
@@ -20,6 +20,11 @@
       "version-string": "2023.07.03.00",
       "port-version": 0
     },
+    {
+      "git-tree": "tofix",
```
Comment (jjerphan) [inline versions/f-/folly.json:24]: Note to reviewer: this needs to be adapted but I have not find how to adapt this.

Comment (c8ef) [reply versions/f-/folly.json:24]: > The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.

**Comment by Thomas1664** [comment]
You can't add an old version. It is only possible to update the port to a newer version.

**Comment by jjerphan** [comment]
Can't I conventionally or technically?

File: versions/f-/folly.json (hunk 1 above)
Comment (jjerphan) [reply versions/f-/folly.json:24]: I currently get this error:

```
error: can't load port allegro5
error: while loading /home/jjerphan/dev/vcpkg/ports/allegro5/vcpkg.json:
$.default-features[0]: mismatched type: expected an identifier
$.default-features[1]: mismatched type: expected an identifierinternal error: /builddir/build/BUILD/vcpkg-tool-2023-06-22/src/vcpkg/commands.add-version.cpp(412): vcpkg has crashed; no additional details are available.
Please open an issue at https://github.com/microsoft/vcpkg/issues/new?template=other-type-of-bug-report.md&labels=category:vcpkg-bug with detailed steps to reproduce the problem.
```

**Comment by Thomas1664** [comment]
> Can't I conventionally or technically?

This is technically impossible.

**Comment by jjerphan** [comment]
Why?

**Pull request closed without merging**